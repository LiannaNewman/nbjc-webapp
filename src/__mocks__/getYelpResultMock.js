const getYelpResultMock = (overrides = {}) => {
  const business = {
    id: '123',
    name: 'La colombe coffee roasters',
    category: 'Coffee',
    address: '924 Blagden Alley Way Washington, D.C. 20001',
    phoneNumber: '(123) 456-7890',
  };
  return {
    ...business,
    ...overrides,
  };
};

export default getYelpResultMock;
