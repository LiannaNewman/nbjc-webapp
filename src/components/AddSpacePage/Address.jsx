import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import useMediaQuery from '@material-ui/core/useMediaQuery';

import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

import { businessProps, addSpaceProps as addSpacePropTypes } from '../../types';
import BusinessResultCard from '../BusinessResultCard';

const styles = (theme) => ({
  listWrapper: {
    display: 'grid',
    gridTemplateColumns: 'repeat(auto-fill, 290px)',
    maxHeight: 460,
    overflow: 'scroll',
    marginBottom: 20,
    [theme.breakpoints.up('mobile')]: {
      maxHeight: 500,
    },
  },
  card: {
    margin: '20px',
  },
  footer: {
    display: 'block',
    marginBottom: 20,
  },
  yelpFooter: {
    marginBottom: 15,
    display: 'flex',
  },
  yelpIcon: {
    width: 45,
  },
  yelpText: {
    lineHeight: '45px',
    marginLeft: 5,
  },
  submitButton: {
    float: 'right',
    marginBottom: 20,
  },
});

const Address = ({
  businessList,
  classes,
  onBack,
  onNext,
  addSpaceProps,
}) => {
  const matches = useMediaQuery('(min-width:376px)');
  const { business: preSelectedBusiness } = addSpaceProps;
  let preSelected;
  if (preSelectedBusiness) {
    preSelected = businessList.findIndex((b) => b.id === preSelectedBusiness.id);
  }
  const [selected, setSelected] = useState(preSelected === undefined ? null : preSelected);
  const onCheck = (index) => {
    if (selected === index) {
      setSelected();
    } else {
      setSelected(index);
    }
  };

  return (
    <>
      <Typography variant={matches ? 'h4' : 'subtitle1'} align="center">Select the location of the space you want to submit to The Lavender Book.</Typography>
      <div className={classes.listWrapper}>
        {businessList.map((business, index) => (
          <BusinessResultCard
            business={business}
            key={business.yelp_id}
            checked={selected === index}
            onCheck={() => onCheck(index)}
            overrideClasses={{ resultCard: classes.card }}
          />
        ))}
      </div>
      <div className={classes.footer}>
        <Button
          type="submit"
          variant="contained"
          color="primary"
          className={classes.submitButton}
          fullWidth={!matches}
          disabled={selected === undefined || selected === null}
          onClick={() => onNext({ business: businessList[selected] })}
          data-testid="addspace-business-next"
          disableElevation
        >
          Next
        </Button>
        <Button
          type="cancel"
          variant="outlined"
          color="primary"
          className={classes.backButton}
          fullWidth={!matches}
          onClick={onBack}
          disableElevation
        >
          Back
        </Button>

      </div>
      <div className={classes.yelpFooter}>
        <img src="/yelp_burst_icon.png" alt="Yelp Burst Icon" className={classes.yelpIcon} />
        <span className={classes.yelpText}>This space information is powered by Yelp.</span>
      </div>
    </>
  );
};

Address.propTypes = {
  businessList: PropTypes.arrayOf(PropTypes.shape(businessProps)).isRequired,
  onNext: PropTypes.func,
  onBack: PropTypes.func,
  addSpaceProps: PropTypes.shape(addSpacePropTypes),
  classes: PropTypes.shape({}).isRequired,
};

Address.defaultProps = {
  onNext: () => {},
  onBack: () => {},
  addSpaceProps: {},
};

export default withStyles(styles)(Address);
