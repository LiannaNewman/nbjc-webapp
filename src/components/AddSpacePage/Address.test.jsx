import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import getYelpResultMock from '../../__mocks__/getYelpResultMock';

import Address from './Address';

describe('Address', () => {
  test('renders with default values', () => {
    const businessList = [...Array(10)].map((_, i) => getYelpResultMock({ id: `${i}` }));
    const onNext = jest.fn();
    const onBack = jest.fn();
    const { asFragment } = render(
      <Address businessList={businessList} onNext={onNext} onBack={onBack} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders with the preselectedBusiness and passes selected business on submit', () => {
    const businessList = [...Array(10)].map((_, i) => getYelpResultMock({ id: `${i}` }));
    const onNext = jest.fn();
    const onBack = jest.fn();
    render(
      <Address
        businessList={businessList}
        onNext={onNext}
        onBack={onBack}
        addSpaceProps={{ business: businessList[0] }}
      />,
    );
    const business1 = screen.getByTestId('businessresultcard-0').querySelector('input');
    expect(business1.checked).toBe(true);
    const business2 = screen.getByTestId('businessresultcard-1').querySelector('input');
    fireEvent.click(business2);
    expect(business1.checked).toBe(false);
    expect(business2.checked).toBe(true);
    const nextButton = screen.getByText('Next').closest('button');
    fireEvent.click(nextButton);
    expect(onNext.mock.calls.length).toBe(1);
    expect(onNext.mock.calls[0][0]).toEqual({ business: businessList[1] });
  });
});
