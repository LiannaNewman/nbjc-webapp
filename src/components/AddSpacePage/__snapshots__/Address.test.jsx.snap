// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address renders with default values 1`] = `
<DocumentFragment>
  <h6
    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter"
  >
    Select the location of the space you want to submit to Lavender Book
  </h6>
  <div
    class="Address-listWrapper-1"
  >
    <div
      class="Address-card-2"
      data-testid="businessresultcard-0"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-5 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-7 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-8 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-13"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Address-card-2"
      data-testid="businessresultcard-1"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-5 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-7 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-8 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-13"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Address-card-2"
      data-testid="businessresultcard-2"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-5 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-7 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-8 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-13"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Address-card-2"
      data-testid="businessresultcard-3"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-5 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-7 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-8 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-13"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Address-card-2"
      data-testid="businessresultcard-4"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-5 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-7 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-8 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-13"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Address-card-2"
      data-testid="businessresultcard-5"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-5 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-7 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-8 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-13"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Address-card-2"
      data-testid="businessresultcard-6"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-5 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-7 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-8 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-13"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Address-card-2"
      data-testid="businessresultcard-7"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-5 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-7 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-8 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-13"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Address-card-2"
      data-testid="businessresultcard-8"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-5 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-7 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-8 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-13"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Address-card-2"
      data-testid="businessresultcard-9"
    >
      <div
        class="MuiPaper-root MuiCard-root BusinessResultCard-card-5 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div>
            <h6
              class="MuiTypography-root BusinessResultCard-title-7 MuiTypography-subtitle1"
            >
              La colombe coffee roasters
            </h6>
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-8 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-13"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Coffee
          </span>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Address:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              924 Blagden Alley Way Washington, D.C. 20001
            </p>
          </div>
          <div
            class="BusinessResultCard-wrapper-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Phone number:
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              (123) 456-7890
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Address-footer-3"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained Address-submitButton-4 MuiButton-containedSecondary MuiButton-disableElevation MuiButton-fullWidth"
      data-testid="addspace-business-next"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Next
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-disableElevation MuiButton-fullWidth"
      tabindex="0"
      type="cancel"
    >
      <span
        class="MuiButton-label"
      >
        Back
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
